File: tests/fixtures/sample-python-project-il/code/app/services/debug.py
import os

# Spec forbids any debug env-var in production builds. PROD_DEBUG gates
# verbose request logging and must not be read.
# IL-DRIFT: ForbiddenArtifact:prod-debug-env / forbidden.env-var.PROD_DEBUG.present
DEBUG_MODE = os.environ.get("PROD_DEBUG") == "true"


def debug_log(*args) -> None:
    if DEBUG_MODE:
        print("[debug]", *args)

File: tests/fixtures/sample-csharp-project-il/code/Services/Debug.cs
// Spec forbids any debug env-var in production builds. PROD_DEBUG gates verbose
// request logging and must not be read.
// IL-DRIFT: ForbiddenArtifact:prod-debug-env / forbidden.env-var.PROD_DEBUG.present
namespace SampleApi.Services;

public static class Debug
{
    private static readonly bool DebugMode =
        Environment.GetEnvironmentVariable("PROD_DEBUG") == "true";

    public static void DebugLog(params object[] args)
    {
        if (DebugMode)
        {
            Console.WriteLine("[debug] " + string.Join(" ", args));
        }
    }
}
